### PR TITLE
backend/enhancement: Remove rate limiting on signature auth

### PR DESCRIPTION
**Pull request by khuzema786**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Removing rate limiting on Signature Authentication

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs [hunk 1]
```diff
@@ -226,10 +227,11 @@ auth isDirectAuth req mbBundleVersion mbClientVersion = do
       else return (Nothing, Nothing, regToken.authType)
   return $ AuthRes regToken.id regToken.attempts authType token personApiEntity
 
-buildPerson :: (EncFlow m r, DB.EsqDBReplicaFlow m r) => AuthReq -> Maybe Version -> Maybe Version -> Id DMerchant.Merchant -> m SP.Person
+buildPerson :: (EncFlow m r, DB.EsqDBReplicaFlow m r, CacheFlow m r) => AuthReq -> Maybe Version -> Maybe Version -> Id DMerchant.Merchant -> m SP.Person
 buildPerson req bundleVersion clientVersion merchantId = do
   pid <- BC.generateGUID
   now <- getCurrentTime
+  merchantConfig <- QMSUC.findByMerchantId merchantId >>= fromMaybeM (MerchantServiceUsageConfigNotFound merchantId.getId)
   personWithSameDeviceToken <- listToMaybe <$> DB.runInReplica (Person.findBlockedByDeviceToken req.deviceToken)
```
Comment (piyushKumar-1) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs:235]: We can avoid this db query as well if fraud detection is disabled ?

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs [hunk 2]
```diff
@@ -187,7 +188,7 @@ auth isDirectAuth req mbBundleVersion mbClientVersion = do
   person <-
     Person.findByRoleAndMobileNumberAndMerchantId SP.USER countryCode mobileNumberHash merchant.id
       >>= maybe (createPerson req mbBundleVersion mbClientVersion merchant.id) return
-  checkSlidingWindowLimit (authHitsCountKey person)
+  unless isDirectAuth $ checkSlidingWindowLimit (authHitsCountKey person)
```
Comment (piyushKumar-1) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs:191]: We dont want rate limiter for paytm ?

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs [hunk 3]
```diff
@@ -226,10 +227,11 @@ auth isDirectAuth req mbBundleVersion mbClientVersion = do
...
   return $ AuthRes regToken.id regToken.attempts authType token personApiEntity
 
-buildPerson :: (EncFlow m r, DB.EsqDBReplicaFlow m r) => AuthReq -> Maybe Version -> Maybe Version -> Id DMerchant.Merchant -> m SP.Person
+buildPerson :: (EncFlow m r, DB.EsqDBReplicaFlow m r, CacheFlow m r) => AuthReq -> Maybe Version -> Maybe Version -> Id DMerchant.Merchant -> m SP.Person
 buildPerson req bundleVersion clientVersion merchantId = do
   pid <- BC.generateGUID
   now <- getCurrentTime
+  merchantConfig <- QMSUC.findByMerchantId merchantId >>= fromMaybeM (MerchantServiceUsageConfigNotFound merchantId.getId)
   personWithSameDeviceToken <- listToMaybe <$> DB.runInReplica (Person.findBlockedByDeviceToken req.deviceToken)
   let isBlockedBySameDeviceToken = maybe False (.blocked) personWithSameDeviceToken
```
Comment (piyushKumar-1) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs:236]: Also can you add this in berify function as well ? Line 350 below.

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs (hunk 2 above)
Comment (khuzema786) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs:191]: Nope, signature authentication will rate limit, people can go back and again open sdk

**Review by piyushKumar-1 — APPROVED** [review]

**Pull request merged**